src/main/java/org/apache/commons/collections4/CollectionUtils.java

```diff
@@ -1112,6 +1093,7 @@ public int hashCode() {
      * @return a boolean indicating whether the collection has changed or not.
      * @throws NullPointerException if the collection or iterator is null
      */
+    @SuppressWarnings("unchecked")
```

Comment:
I agree with you. You are right. I had try to compile the code by `javac` with flag `-Xlint:all` and did not see any warnings. Then Intellij Idea shows false warning or the warning are not for compiler, it is more for a programmer. 
Yes, all good to be merged.